specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscCompilationJob.json

```diff
@@ -383,8 +383,9 @@
           "x-nullable": false
         },
         "provisioningState": {
-          "$ref": "#/definitions/JobProvisioningStateProperty",
-          "description": "The current provisioning state of the job."
+          "type": "string",
+          "description": "The current provisioning state of the job.",
```

Comment:
@ravbhatnagar, mind confirming whether or not this is the correct LRO format?

Reply:
@vrdmr - why are you moving away from this being an enum?

Reply:
We are seeing the following error when using the generated SDK in the cmdlet work we are doing - And we wanted to avoid this with minimal change to the whole process.

```
DEBUG: AzureQoSEvent: CommandName - Start-AzureRmAutomationRunbook; IsSuccess - False; Duration - 00:00:13.0451276; Exception - Microsoft.Rest.SerializationException: Unable to deserialize the response. ---> Newtonsoft.Json.JsonSerializationException: Error converting
value "Processing" to type 'Microsoft.Azure.Management.Automation.Models.JobProvisioningStateProperty'. Path '', line 1, position 427. ---> System.ArgumentException: Could not cast or convert from System.String to
Microsoft.Azure.Management.Automation.Models.JobProvisioningStateProperty.
   at Newtonsoft.Json.Utilities.ConvertUtils.EnsureTypeAssignable(Object value, Type initialType, Type targetType)
   at Newtonsoft.Json.Utilities.ConvertUtils.ConvertOrCast(Object initialValue, CultureInfo culture, Type targetType)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)
```

Reply:
@vrdmr, So this PR is a workaround for a C# generator issue that you're experiencing? Mind if we try to have the C# folks help you instead of removing some of the hints that we give developers as they try to interact with your service?

Reply:
@shahabhijeet @dsgouda Any ideas on why C# gives error while casting?

FYI @safeermohammed

Reply:
@vrdmr is there a repro by any chance in the SDK repo tests or is this something observed only with powershell cmdlets

Reply:
Well, looking at this, they were declaring the `provisioningState` property as a `object` of type `JobProvisioningStateProperty` that has a property called `provisioningState`.

This has nothing to do with making it an `enum`, this has to do with the fact that `provisioningState` was incorrectly declared in the first place.

Before someone goes off and stuffs in another incorrect model, why not try fixing the problem, by making `JobProvisioningStateProperty` just an string with an enum?

``` json

"JobProvisioningStateProperty": {	
      "type": "string",
      "description": "The provisioning state of the resource.",	
      "enum": [	
        "Failed",	
        "Succeeded",	
        "Suspended",	
        "Processing"	
      ],	
      "x-ms-enum": {	
        "name": "JobProvisioningState",	
        "modelAsString": true	
      }	
     }
```

Reply:
@fearthecowboy  thanks man. This seems to be the fix to my problem [here](https://github.com/Azure/azure-sdk-for-net/issues/4399)

When can we expect this to be released as a nuget package 3.0.4? (I'm not really familiar with the release cycle/process)

It's a breaking bug in the 3.0.3 preview for many methods using the deserialize

Also - is it possible for the Json converter to just make the values null if it fails to deserialize instead of the whole response body fails? Like a TryGetValue thing :)

From Operations.cs:
```
// Deserialize Response
            if ((int)_statusCode == 200)
            {
                _responseContent = await _httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
                try
                {
                    _result.Body = Rest.Serialization.SafeJsonConvert.DeserializeObject<Page1<Operation>>(_responseContent, Client.DeserializationSettings);
                }
                catch (JsonException ex)
                {
                    _httpRequest.Dispose();
                    if (_httpResponse != null)
                    {
                        _httpResponse.Dispose();
                    }
                    throw new SerializationException("Unable to deserialize the response.", _responseContent, ex);
                }
            }
```

Reply:
@MortenMeisler Please open a PR with the regenerated code against the Net SDK repo and once that's merged, the Nuget can be published immediately.
Versioning depends on whether it is a breaking change in the SDK and will be addressed in the PR against the SDK repo itself.